Core/ExpectedException/ExpectedExceptionRewriter.cs

```diff
@@ -0,0 +1,67 @@
+﻿#region copyright
+
+// 
+// Copyright (c) rubicon IT GmbH
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace NUnit2To3SyntaxConverter.ExpectedException
+{
+  public class ExpectedExceptionRewriter : CSharpSyntaxRewriter
+  {
+    private readonly NUnitAssemblyFilter _assemblyFilter;
+    private readonly ISyntaxTransformer<MethodDeclarationSyntax, ExpectedExceptionModel> _attributeRemover;

```

Comment:
Could you use the interface type `IExpectedExceptionModel` in this file (except in return types)?

Reply:
Actually, even though it's probably irrelevant, as a thought experiment: The way this class is currently written, we probably should't use `ISyntaxTransformer<MethodDeclarationSyntax, ExpectedExceptionModel>` as the types of the fields, since not any `ISyntaxTransformer<MethodDeclarationSyntax, ExpectedExceptionModel>` would do because we're  using them in a very specific way. 

For true loose coupling, we'd need to store an IEnumerable of `ISyntaxTransformer<MethodDeclarationSyntax, ExpectedExceptionModel>` and apply their transformations one by one, and since that wouldn't make sense here it's perhaps better to store the more concrete type of the the `ISyntaxTransformer`s